Add 0.4.11 release notes (#3213)
Napari/napari: docs/release/release_0_4_11.md

# napari 0.4.11

We're happy to announce the release of napari 0.4.11!
napari is a fast, interactive, multi-dimensional image viewer for Python.
It's designed for browsing, annotating, and analyzing large multi-dimensional
images. It's built on top of Qt (for the GUI), vispy (for performant GPU-based
rendering), and the scientific Python stack (numpy, scipy).

For more information, examples, and documentation, please visit our website:
https://github.com/napari/napari

## Highlights

> 

Reviewer: ```suggestion
We've known for a while that 3D interactivity is one of the areas where napari needed improvement (#515). This release introduces ways to use the mouse to interact with data in 3D (#3037). Features like label picking (#3074) and label painting/erasing (#3108) in 3D are just the beginning and we're excited to see where this takes us! For more details, please see the documentation at [https://napari.org/guides/stable/3D_interactivity.html](https://napari.org/guides/stable/3D_interactivity.html).

Our volume rendering functionality has been significantly improved and now includes the ability to render arbitrary planes through volumes (#3023) and add clipping planes to restrict rendering to a region of interest (#3140). For now, these features are marked as `experimental` and the API around their use is likely to change in future versions of napari.

Last but not least, some common operations are now much more accessible from the GUI thanks to a new context menu on the layer list (#2556) and buttons for controlling how autoscaling is performed for images (#3022).
```

## New Features

> - Add context menu on layer list, introduce `QtActionContextMenu`. (#2556)

Reviewer 2: highlight?

Reviewer: added

- Add activity dialog and style progress bars (#2656)
- Add playback options to settings (#2933)
- Add autoscale modes to image layer model, and buttons to GUI (#3022)
- Arbitrary plane rendering prototype (#3023)

> - Add view ray and labels selection in 3D (#3037)

Reviewer 2: highlight, obv.

- Grid mode popup (#3084)
- Add 3D fill, "mill", and "print" on top of #3074 (#3108)
- Add positive tail length to tracks layer (#3138)

> - Arbitrary clipping planes for volumes in the image layer (#3140)

Reviewer 2: highlight?

Reviewer: added

- Mask image from points layer (#3151)

## Improvements
- Add `assign_[plugin]_to_extension` methods on plugin_manager.  (#2695)
- Use QDoubleRangeSlider from superqt package (#2752)
- Use labeled sliders from superqt (#2753)
- Shortcuts UI (#2864)
- Convert TextManager to an EventedModel (#2885)
- Make maximum brush size flexible (#2901)
- Allow layer to register action on double clicks. (#2907)
- Reduce numpy array traceback print (#2910)
- Provide manual deepcopy implementation for translations strings. (#2913)
- Make Points construction with properties consistent with setting properties (#2916)
- Add search field to plugin dialog  (#2923)

> - Refactor settings manager to allow setting all preferences with env vars and context managers (#2932)

Reviewer 2: might want to mention this... though it's mostly a refactor, the "feature" is the ability to now set anything with env vars

Author: moved to feature

- Add initital support to install from conda/mamba (#2943)
- Shape Mouse refactor (#2950)
- Make handling properties more consistent across some layer types (#2957)
- Labels paintbrush now takes anisotropy into account (#2962)
- Remove mode guards for selection interactions in points (#2982)
- Emit data event when moving, adding or removing shapes or points (#2992)
- Add TypedMutableMapping and EventedDict (#2994)
- Add isosurface rendering to Labels (#3006)
- Remove mentions of _mode_history (2987) (#3008)
- Change opacity slider to float slider (#3016)

> - Add projections to layer context menu, allow grouping and nesting of menu items (#3028)

Reviewer 2: move to feature?

- Refactor the Point, Label and Shape Layer Mode logic. (#3050)
- Make flash effect feel more instant (#3060)
- Use enum values internally for settings. (#3063)
- Update vendored volume visual from vispy (#3064)

> - Skip progress indicator test when viewer is not shown (#3065)

Reviewer 2: move to tests

Author: done

- Allow for multiple installs and update buttons to reflect state (#3067)
- Unify plugin wording and installer dialog to display only package/plugins (#3071)

> - 3D label picking and label ID in status bar (#3074)

Reviewer 2: feature?  highlight?

Reviewer: added

- Store unmaximized size if napari closes maximized (#3075)

> - Add `add_<shape_type>` method for each shape type (#3076)

Reviewer 2: feature?

Author: done

- Change shapes default edge color to middle gray (#3113)
- Change default text overlay color to mid grey (#3114)
- Add 3D get_value to Shapes (#3117)
- Replace custom signals to accept/reject (#3120)
- Remove old utils/settings/constants file (#3122)
- NAPARI_CATCH_ERRORS disable notification manager (#3126)
- Save files after launch from ipython (#3130)
- Don't try to read unknown formats in builtin reader plugin (#3145)
- Make current viewer accessible from the napari module (#3149)
- Rename `Layer.plane` to `Layer.slicing_plane` (#3150)
- Update new label action to work with tensorstore arrays (#3153)
- Always raise PluginErrors (#3157)
- Establish better YAML encoding for settings (fix enum encoding issue). (#3163)

> - Don't clear blocked plugins when closing preferences dialog (#3164)

Reviewer 2: bug fix

Author: moved

- Revert `Points` `remove_selected` always overwriting `self._value` to `None` (#3165)
- Fix window geometry loading bug, and make `ApplicationSettings` types more accurate (#3182)
- Fix missing import in napari.__init__.pyi (#3183)
- Fix incorrect window position storage (#3196)
- Fix incorrect use of dims_order when 3D painting (#3202)

## API Changes
- Remove brush shape (#3047)
- Enforce layer.metadata as dict (#3020)
- Use enum objects in EventedModel (#3112)

## UI Changes
- Remove keybindings dialog from help menu (#3048)
- Remove plugin sorter from plugin install dialog (#3069)
- Update Labels layer keybindings to be more ergonomic (#3072)

## Deprecations

## Build Tools, Tests, Documentation, and other Tasks
- Add imagecodecs to the bundle to open additional tiffs (#2895)
- Make ordering of releases manual (#2921)
- Add alister burt to team page (#2937)
- Use briefcase 0.3.1 on all platforms (#2980)
- Speedup one of the slowest test. (#2997)
- Update plugin guide with references and instructions for napari-hub (#3055)
- Add missing libraries in docker file and entrypoint (#3081)
- Update documentation regarding the hub (#3091)

> - Fix stubgen and package stubs in wheel/sdists (#3105)

Reviewer 2: this is actually a reasonably big quality of life improvement for those using napari in their IDE.  maybe feature?

Author: moved

images. It's built on top of Qt (for the GUI), vispy (for performant GPU-based
rendering), and the scientific Python stack (numpy, scipy).

For more information, examples, and documentation, please visit our website:
https://github.com/napari/napari

## Highlights
We've known for a while that 3D interactivity is one of the areas where napari needed improvement (#515). This release introduces ways to use the mouse to interact with data in 3D (#3037). Features like label picking (#3074) and label painting/erasing (#3108) in 3D are just the beginning and we're excited to see where this takes us! For more details, please see the documentation at [https://napari.org/guides/stable/3D_interactivity.html](https://napari.org/guides/stable/3D_interactivity.html).

Our volume rendering functionality has been significantly improved and now includes the ability to render arbitrary planes through volumes (#3023) and add clipping planes to restrict rendering to a region of interest (#3140). For now, these features are marked as `experimental` and the API around their use is likely to change in future versions of napari.

> Last but not least, some common operations are now much more accessible from the GUI thanks to a new context menu on the layer list (#2556) and buttons for controlling how autoscaling is performed for images (#3022).

Reviewer: ```suggestion
Last but not least, some common operations are now much more accessible from the GUI thanks to a new context menu on the layer list (#2556 and #3028) and buttons for controlling image contrast limit scaling (#3022).
```

## New Features
- Add context menu on layer list, introduce `QtActionContextMenu`. (#2556)
- Add activity dialog and style progress bars (#2656)
- Add playback options to settings (#2933)
- Add autoscale modes to image layer model, and buttons to GUI (#3022)
- Arbitrary plane rendering prototype (#3023)
- Add view ray and labels selection in 3D (#3037)
- Grid mode popup (#3084)
- Add 3D fill, "mill", and "print" on top of #3074 (#3108)
- Add positive tail length to tracks layer (#3138)
- Arbitrary clipping planes for volumes in the image layer (#3140)
- Mask image from points layer (#3151)

> 

Reviewer: ```suggestion
- Add projections to layer context menu, allow grouping and nesting of menu items (#3028)
```

- Use QDoubleRangeSlider from superqt package (#2752)
- Use labeled sliders from superqt (#2753)
- Shortcuts UI (#2864)
- Convert TextManager to an EventedModel (#2885)
- Make maximum brush size flexible (#2901)
- Allow layer to register action on double clicks. (#2907)
- Reduce numpy array traceback print (#2910)
- Provide manual deepcopy implementation for translations strings. (#2913)
- Make Points construction with properties consistent with setting properties (#2916)
- Add search field to plugin dialog  (#2923)
- Refactor settings manager to allow setting all preferences with env vars and context managers (#2932)
- Add initital support to install from conda/mamba (#2943)
- Shape Mouse refactor (#2950)
- Make handling properties more consistent across some layer types (#2957)
- Labels paintbrush now takes anisotropy into account (#2962)
- Remove mode guards for selection interactions in points (#2982)
- Emit data event when moving, adding or removing shapes or points (#2992)
- Add TypedMutableMapping and EventedDict (#2994)
- Add isosurface rendering to Labels (#3006)
- Remove mentions of _mode_history (2987) (#3008)
- Change opacity slider to float slider (#3016)

> - Add projections to layer context menu, allow grouping and nesting of menu items (#3028)

Reviewer: ```suggestion
```
